apps/jobs/views.py

```diff
@@ -318,16 +378,70 @@ def create(self, request, *args, **kwargs):
                         status=status.HTTP_406_NOT_ACCEPTABLE,
                     )
 
-        ## Save the data into the database
-        # Update the fields
-        serializer = self.get_serializer(data=request.data)
-        serializer.is_valid(raise_exception=True)
-        self.perform_create(serializer)
-        data = serializer.validated_data
-        headers = self.get_success_headers(data)
-        return Response(
-            serializer.data, status=status.HTTP_201_CREATED, headers=headers
-        )
+        # Check if the given user_id actually belongs to the user
+        # This check is necessary because only specific users can update their profile
+
+        # get the user_id from the url path
+        if user_id := search("\w{32}", request.stream.path):
+            user_id = user_id[0]
+
+        # if case, if someone crafts a request in such a way that
+        # user_id in accessToken and user_id in the api path are same
+
+        # perform check on database as well
+        if user_id == payload["user_id"]:
```

Comment:
we dont have to check, fetch it from payload itself

Reply:
Removed this comparison.